### PR TITLE
Add GH action for publishing docs

### PR DESCRIPTION
**Pull request by lorensr**
**What changed?**

Action publishes [grpc docs](https://api-docs.temporal.io/) on push to master.

**Why?**

So I don't have to manually do it.

Tested with:

```
act -W .github/workflows/publish-docs.yml -s VERCEL_TOKEN=foo -s VERCEL_PROJECT_ID=bar -s VERCEL_ORG_ID=baz
```

Added necessary secrets under repo settings.

### CODE REVIEW

File: .github/workflows/publish-docs.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      - name: Generate docs
+        run: |
+          docker run -v $(pwd)/docs:/out -v $(pwd)/:/protos pseudomuto/protoc-gen-doc --doc_opt=html,index.html $(find temporal/api -type f -name "*.proto")
+      - name: Deploy
+        run: npx vercel deploy docs/ --prod --token=${{ secrets.VERCEL_TOKEN }}
```
Comment (mindaugasrukas) [inline .github/workflows/publish-docs.yml:23]: Is `npx` installed by default? Shouldn't it be installed first?

Comment (lorensr) [reply .github/workflows/publish-docs.yml:23]: It's included with node, which is installed by default: https://github.com/actions/runner-images#software-and-image-support

**Comment by lorensr** [comment]
@mjameswh done

**Review by mjameswh — APPROVED** [review]
LGTM

**Review by mindaugasrukas — APPROVED** [review]

**Pull request merged**